app/src/main/java/com/eveningoutpost/dexdrip/watch/miband/MiBandService.java

```diff
@@ -164,20 +151,120 @@ public QueueMessage(String functionName, String message_type, String message, St
         startBgTimer();
     }
 
+    private static final boolean isBetweenValidTime(Date startTime, Date endTime, Date currentTime) {
+        //Start Time
+        Calendar StartTime = Calendar.getInstance();
+        StartTime.setTime(startTime);
+        StartTime.set(1, 1, 1);
+
+        Calendar EndTime = Calendar.getInstance();
+        EndTime.setTime(endTime);
+        EndTime.set(1, 1, 1);
+
+        //Current Time
+        Calendar CurrentTime = Calendar.getInstance();
+        CurrentTime.setTime(currentTime);
+        CurrentTime.set(1, 1, 1);
```

Comment:
I'm not sure if this works correct at midnight when the day switches. `currentTime` already contains the date.